sdk/nodejs/cmd/run/run.ts

```diff
@@ -314,6 +324,10 @@ ${defaultMessage}`);
                 programExport = require(program);
             }
 
+            if (await containsTSAndJSModules(program)) {
+                log.warn("Found a TypeScript project containing an index.js file and no explicit entrypoint in Pulumi.yaml - Pulumi will use index.js");
```

Comment:
Maybe I'm just missing it, but I'm not seeing how "main" is being checked before printing this?

Reply:
Ah - in the checking function above, we branch on whether the `program` we're being handed is a directory or not.  If it's a directory, we're definitely doing some implicit resolution things.  If not, they've been explicit enough in their `main` definition that we likely don't need to warn them.

Reply:
Ah ok I see now, we always call this but this only does anything if program is a directory.